jerry-core/lit/lit-magic-strings.c

```diff
@@ -197,18 +199,40 @@ bool lit_is_ex_utf8_string_magic (const lit_utf8_byte_t *string_p, /**< utf-8 st
 /**
  * Compare utf-8 string and magic string for equality
  *
- * @return true if strings are equal
- *         false otherwise
+ * @return 0, if areas are equal;
+ *         <0, if string's content is lexicographically less, than magic string's content;
+ *         >0, otherwise
  */
-bool
+int
 lit_compare_utf8_string_and_magic_string (const lit_utf8_byte_t *string_p, /**< utf-8 string */
```

Comment:
There is two crucial difference, which requires the existence of a function:
-  the input string can contains \0, which is allowed.
- The actual size of the input string is different from the received. We only extracts a part of this string, and received the size of it.